Sources/Build/BuildPlan.swift

```diff
@@ -1277,6 +1277,19 @@ public final class ProductBuildDescription {
             let stdlib = buildParameters.toolchain.macosSwiftStdlib
             args += ["-Xlinker", "-rpath", "-Xlinker", stdlib.pathString]
           }
+
+          // When deploying to macOS prior to macOS 12, add an rpath to the
+          // back-deployed concurrency libraries.
+          if buildParameters.triple.isDarwin(),
+             let macOSSupportedPlatform = product.targets[0].underlyingTarget.getSupportedPlatform(for: .macOS),
```

Comment:
Just a note to anyone else reading that the `[0]` should be safe since we've already checked for targetless products before this.